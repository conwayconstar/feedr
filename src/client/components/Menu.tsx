import React from 'react';
import { DietarySelection } from 'feedr';

export type MenuProps = {
  count: number;
  dietarySelection: DietarySelection;
};

const Menu: React.FC<MenuProps> = ({ count, dietarySelection }) => (
  <div className="menu-summary">
    <div className="container">
      <div className="row">
        <div className="col-6 menu-summary-left">
          <span className="count">{`${count} item${count !== 1 ? 's' : ''}`}</span>
        </div>
        <div className="col-6 menu-summary-right">
          {Object.keys(dietarySelection).map((dietary) => (
            <span key={dietary}>
              <span className="dietary-count">{`${dietarySelection[dietary]}x`}</span>
              <span className="dietary">{dietary}</span>
            </span>
          ))}
        </div>
      </div>
    </div>
  </div>
);

export default Menu;
