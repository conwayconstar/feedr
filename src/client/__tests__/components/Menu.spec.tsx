import { shallow } from 'enzyme';
import React from 'react';
import Menu, { MenuProps } from '../../components/Menu';

describe('Menu component', () => {
  const props: MenuProps = {
    count: 4,
    dietarySelection: {
      ve: 2,
      v: 3,
      df: 4,
    },
  };
  const wrapper = shallow(<Menu {...props} />);

  it('Menu should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Menu count label should be 4 items', () => {
    expect(wrapper.find('.count').text()).toEqual('4 items');
  });

  it('Menu count label should be 1 item', () => {
    wrapper.setProps({
      count: 1,
      item: {
        id: 10010,
        name: 'Pasta Salad Box',
        dietaries: ['ve', 'v', 'gf', 'df'],
      },
    });
    wrapper.update();
    expect(wrapper.find('.count').text()).toEqual('1 item');
  });
});
