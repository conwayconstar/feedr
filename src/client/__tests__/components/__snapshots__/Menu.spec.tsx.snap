// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component Menu should match snapshot 1`] = `
<div
  className="menu-summary"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-6 menu-summary-left"
      >
        <span
          className="count"
        >
          4 items
        </span>
      </div>
      <div
        className="col-6 menu-summary-right"
      >
        <span
          key="ve"
        >
          <span
            className="dietary-count"
          >
            2x
          </span>
          <span
            className="dietary"
          >
            ve
          </span>
        </span>
        <span
          key="v"
        >
          <span
            className="dietary-count"
          >
            3x
          </span>
          <span
            className="dietary"
          >
            v
          </span>
        </span>
        <span
          key="df"
        >
          <span
            className="dietary-count"
          >
            4x
          </span>
          <span
            className="dietary"
          >
            df
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
